### ISSUE TITLE
AzureRM - issue with Location and Availability sets

### ISSUE THREAD

**Issue by dougt1** [hashicorp/terraform#8183]
**Version**
 v 0.7.0 

**Affected Resource**
azurerm_availability_set

**Issue Details**

I have an availability set in Azure (Resource Manager) with 2 VM's. On every run my location is seen as different (as appears case sensitive) - this results in the forced creation of new resources within the availability set - resulting in the deletion and re-creation of all vm's in the availability set. It doesn't seem to matter what case I specify the location as - on subsequent runs it always returns as a change and wants to recreate the resource group and the VM's within the group.

Results from a terraform plan below:-

```
-/+ module.availability_set_test.azurerm_availability_set.availability_set
location: "UKSouth2" => "uksouth2" (forces new resource)
name: "availability-set-test" => "availability-set-test"
platform_fault_domain_count: "3" => "3"
platform_update_domain_count: "5" => "5"
resource_group_name: "test-prod-rg" => "test-prod-rg"
tags.%: "1" => "1"
tags.environment: "production" => "production"
```

Using a proxy between Terraform and Azure i can see that Azure responds with location in lowercase for most resources:

```
    "kind": "Storage",
    "location": "uksouth2",
    "name": "testdisk007",

    "type": "Microsoft.Network/networkSecurityGroups",
    "location": "uksouth2",
```

however for Availability sets it returns the location as:-

```
    "name": "availability-set-test",
    "type": "Microsoft.Compute/availabilitySets",
    "location": "UKSouth2",
```

This causes Terraform to recreate the Availability set and also the VM's within the availability set on every plan/apply.

I did not see the same behavior on version 0.6.16

**Comment by stack72** [comment]
Thanks @dougt1 

Will get this looked into!

Paul

**Comment by dstokes1** [comment]
I am working with @dougt1 on this, has there been any update on the issue?

**Comment by titanous** [comment]
I pushed a fix for this in #9208.

**Comment by pmcatominey** [comment]
@dougt1 this was fixed in #10409, any casing changes to the location field are suppressed in the diff now.

**Comment by stack72** [comment]
Closed via #10409
